core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -372,45 +373,32 @@ public FormIndex deleteRepeat(FormIndex index) {
     }
 
     public void createNewRepeat(FormIndex index) throws InvalidReferenceException {
-        TreeReference destRef = getChildInstanceRef(index);
-        TreeElement template = mainInstance.getTemplate(destRef);
+        TreeReference repeatContextRef = getChildInstanceRef(index);
+        TreeElement template = mainInstance.getTemplate(repeatContextRef);
 
-        mainInstance.copyNode(template, destRef);
+        mainInstance.copyNode(template, repeatContextRef);
 
-        preloadInstance(mainInstance.resolveReference(destRef));
+        preloadInstance(mainInstance.resolveReference(repeatContextRef));
 
         // Fire jr-insert events before "calculate"s
-        Vector<Triggerable> triggeredDuringInsert = new Vector<Triggerable>();
-        processInsertAction(destRef, triggeredDuringInsert);
+        triggeredDuringInsert = new Vector<Triggerable>();
+        triggerActionsFromEvent(Action.EVENT_JR_INSERT, this, repeatContextRef, this);
```

Comment:
Refactored to allow EVENT_JR_INSERT to utilize the existing framework, instead of calling its own custom `processInsertAction` method